Add initial governance documents (#1213)
sirius-db/sirius: GOVERNANCE.md

# Sirius Governance

> Sirius is a _GPU-native composable analytics engine_ under the Apache-2.0 license. Governance is intentionally minimal and will formalize as the project grows, with the goal of eventually moving under a neutral open source foundation.

Reviewer: ```suggestion
Sirius is a _GPU-native composable analytics engine_ under the [Apache-2.0 license](./LICENSE). Governance is intentionally minimal and will formalize as the project grows, with the goal of eventually moving under a neutral open source foundation.
```

## Roles

> - **Contributors**. Anyone can open issues or pull requests. See [CONTRIBUTING.md](./CONTRIBUTING.md).

Reviewer: And review? Those reviews won't unblock the required review check, but they help the reviewer to understand, PR authors and other reviewers.

The section below about becoming a maintainer also mentions reviews:

> 1. A maintainer nominates a contributor with a short rationale and links to representative work (reviews and design input, not only authored PRs).

```suggestion
- **Contributors**. Anyone can open and contribute to issues and pull requests. See [CONTRIBUTING.md](./CONTRIBUTING.md).
```

> - **Maintainers** are trusted contributors who can review and merge contributions, make technical decisions, manage the maintainer base, and enforce the [CODE_OF_CONDUCT.md](./CODE_OF_CONDUCT.md). They decide by consensus, falling back to a majority vote of maintainers when necessary. See [MAINTAINERS.md](./MAINTAINERS.md).

Reviewer: ```suggestion
- **Maintainers** are trusted contributors who can approve and merge contributions, make technical decisions, manage the maintainer base, and enforce the [CODE_OF_CONDUCT.md](./CODE_OF_CONDUCT.md). They decide by consensus, falling back to a majority vote of maintainers when necessary. See [MAINTAINERS.md](./MAINTAINERS.md).
```

Reviewer 2: In that case, contributors should add "review" right ? because providing review is also a contribution where they can't approve but yet provide feedback.

## Becoming a Maintainer

> 1. A maintainer nominates a contributor with a short rationale and links to representative work (reviews and design input, not only authored PRs).

Reviewer: Maybe we should specify where we communicate this? A public or private channel?

Reviewer 3: I'm thinking opening a PR as a nomination?

Reviewer 2: issue or PR assignment ? github has those (PR ones are often unused)

Reviewer 3: issue also works

> 2. Lazy consensus over five working days — silence counts as approval.

Reviewer: Maybe we should specify a minimum number of votes too?

3. If a concern can't be resolved, a majority vote of maintainers decides.
4. On approval, a pull request adds the person to [MAINTAINERS.md](./MAINTAINERS.md) and access is granted.

Signals we look for: sustained contribution, sound judgment in review and design discussion, and reliability.

## Leaving

Maintainers may step down at any time by opening a pull request removing
themselves from [MAINTAINERS.md](./MAINTAINERS.md).
Maintainers who haven't been active for a while may be moved to emeritus
in [MAINTAINERS.md](./MAINTAINERS.md) by other maintainers, and can return

> by asking.

Reviewer: I'm curious if we need this, when we have the self-removal rule above. If we want this we should probably define "for a while".

Reviewer 3: So the reason we need this is because when someone is no longer active, they might not be even active enough to remove themselves on their own

Reviewer 3: For the period, maybe one year is reasonable?

In rare cases, a maintainer may be removed by a majority vote of the other

> maintainers.

Reviewer: We should maybe specify or give examples of when this applies?

Reviewer 2: Despite being rare, they can also be removed for non-project related reasons too.
